File: pset1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    //Ask for wanted change
    float dollars;
    do
    {
        dollars = get_float("Change owed:");
    }
    while (dollars < 0);
    // Converting dollars to cents
    int cents = round(dollars * 100);
    // making a variable for change left
    int change_left = cents;
    // making a variable forr counting coins
    int coins = 0;
    // cheking for quarters
    while (change_left >= 25)
    {
        change_left = change_left - 25;
        coins++;
    }
    // checking for dimes
    while (change_left >= 10)
    {
        change_left = change_left - 10;
        coins++;
    }
    //checking for nickels
    while (change_left >= 5)
    {
        change_left = change_left - 5;
        coins++;
    }
    //checking for pennies
    while (change_left > 0)
    {
        change_left = change_left - 1;
        coins++;
    }
    //printing the result
    printf("%i\n", coins);
}

File: pset6/cash/cash.py
from cs50 import get_float
dollars = -1.0
while dollars < 0:
    dollars = get_float("Change owed: ")
# converting dollars to cents
cents = round(dollars * 100)
coins = 0
# cheking for quaters
while cents >= 25:
    cents -= 25
    coins += 1
# checking for dimes
while cents >= 10:
    cents -= 10
    coins += 1
# checking for nickels
while cents >= 5:
    cents -= 5
    coins += 1
# checking for pennies
while cents > 0:
    cents -= 1
    coins += 1
# print the result
print(coins)